### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by shrey-creator**

### CODE REVIEW

File: dags/githubTasks/task2CommitHistoryInformation.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def getCommitsInfoGithub():
     repos = responseService.getResponse(urlService.getRepositoriesByOrgNameUrl(org_name),headers)
     since_date_time = datetime.datetime(2022, 3, 1, 0, 0, 0) .isoformat()
```
Comment (shrey-creator) [inline dags/githubTasks/task2CommitHistoryInformation.py:18]: Comment 1

File: dags/githubTasks/task3PRHistoryInformation.py [hunk 2]
```diff
@@ -18,24 +18,25 @@ def getPrHistoryInfoGithub():
     for repo in repos:
         repo_name=repo['name']        
         # merged PR are automatically closed
-        closed_pr_response = responseService.getResponse(urlService.getPRByOrgNameAndRepoNameAndState(org_name,repo_name,'closed'),headers)
+        closed_pr_response = responseService.getResponse(urlService.getPRByOrgNameAndRepoNameAndState(org_name,repo_name,'open'),headers)
         
         for closed_pr in closed_pr_response:
```
Comment (shrey-creator) [inline dags/githubTasks/task3PRHistoryInformation.py:23]: Comment 2

Comment (shrey-creator) [reply dags/githubTasks/task3PRHistoryInformation.py:23]: Reply to comment 2

File: dags/githubTasks/task4BranchHistoryInformation.py [hunk 3]
```diff
@@ -20,30 +20,27 @@ def getBranchHistoryInfoGithub():
...
             is_pr_open=False
             for closed_pr in closed_pr_response:
-                merged_at =closed_pr['merged_at']
-                if(merged_at is None):
-                    is_pr_open=True
-            if(is_pr_open is False and len(closed_pr_response)!=0):
+                response_status_code=responseService.getResponseStatusCode(closed_pr['url']+'/merge',headers)
+                if(response_status_code==204):
+                    is_pr_open=False
+            if(is_pr_open is False and len(closed_pr_response)!=0):     #IF any pr was open and it was merged only then branch is considered as merged
```
Comment (testUserPoc1) [inline dags/githubTasks/task4BranchHistoryInformation.py:32]: Comment 4

File: dags/githubTasks/task4BranchHistoryInformation.py [hunk 4]
```diff
@@ -52,9 +49,14 @@ def getBranchHistoryInfoGithub():
             if time_difference_days > 30:
                 is_branch_stale=True
```
Comment (testUserPoc1) [inline dags/githubTasks/task4BranchHistoryInformation.py:50]: Comment 5

**Review by testUserPoc1 — CHANGES_REQUESTED** [review]

File: dags/githubTasks/task4BranchHistoryInformation.py (hunk 3 above)
Comment (shrey-creator) [reply dags/githubTasks/task4BranchHistoryInformation.py:32]: Reply to comment 4

File: dags/githubTasks/task4BranchHistoryInformation.py (hunk 4 above)
Comment (shrey-creator) [reply dags/githubTasks/task4BranchHistoryInformation.py:50]: Reply to comment 5

**Pull request merged**